cves/CVE-2017-12616.yml

```diff
@@ -217,8 +217,7 @@ mistakes:
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
   answer: |
-    Not enough testing to be able to catch the unexpected behavior. The fix
-    is just to upgrade to the latest release, which seems like a reasonable
-    solution since the vulnerability patch requires at most a minor release
-    upgrade. The thing that made this vulnerability particularly bad is that
-    since it gave access to source code it opened the JSPs to further attacks.
+    The lack of unit testing seems to be the biggest mistake, specifically due to the lack
+    of testing regarding the VirualDirContext class prior to finding the vulnerability.
+    The best way to resolve this issue is to implement a patch that stresses further 
+    checking of requests.
```

Comment:
> The best way to resolve this issue is to implement a patch that stresses further 
    checking of requests.
Is this what they ended up doing as the fix? Unclear as to what the fix was actually implemented as a result of this vulnerability.

Reply:
Added what they did to fix the vulnerability.